paddle/parameter/Parameter.h

```diff
@@ -324,6 +324,7 @@ class Parameter {
   std::vector<std::shared_ptr<IParameterUpdaterHook>> updaterHooks_;
 
 public:
+  void setSharedCount(int cnt) { sharedCount_ = cnt; }
```

Comment:
It seems that `shared count` should be separated from `paddle::Parameter`. Because if we support train multiple neural networks(i.e. topologies) in the one process, we should let `GradientMachine` to maintain shared count? Each `GradientMachine` would handle one neural network and `Parameter's shared count` is used inside one neural network.

Moreover, should we remove the logic about backward callback or pipeline in communication? It gives us not much performance gain but makes our this logic a little bit complicated.